packaging/windows/install_script.nsis

```diff
@@ -0,0 +1,237 @@
+# These will be displayed by the "Click here for support information" link in "Add/Remove Programs"
+!define HELPURL "https://github.com/grafana/agent" # "Support Information" link
+!define UPDATEURL "https://github.com/grafana/agent" # "Product Updates" link
+!define ABOUTURL "https://github.com/grafana/agent" # "Publisher" link
+!include LogicLib.nsh
+
+RequestExecutionLevel admin ;Require admin rights on NT6+ (When UAC is turned on)
+ 
+InstallDir "$PROGRAMFILES\${APPNAME}"
+ 
+# This will be in the installer/uninstaller's title bar
+Name "${APPNAME}"
+Icon "logo.ico"
+outFile "/app/grafana-agent-installer.exe"
+ 
+!include LogicLib.nsh
+!include nsDialogs.nsh
+
+; Everything must be global VARS
+
+# These are handles to dialogs
+Var Dialog
+Var RemoteWriteLabel
+Var RemoteWriteText
+
+Var RemoteWriteUserLabel
+Var RemoteWriteUserText
+
+Var RemoteWritePasswordLabel
+Var RemoteWritePasswordText
+
+Var EnableExporterLabel
+Var EnableExporterCheck
+
+# These are strings that hold the values
+
+Var RemoteUrl
+Var RemoteUser
+Var RemotePassword
+Var EnableExporterValue
+ 
+page directory
+Page custom remoteWriteInfo remoteWriteInfoLeave
+Page custom enableWindowsExporter enableWindowsExporterLeave
+Page instfiles
+
+!macro VerifyUserIsAdmin
+UserInfo::GetAccountType
+pop $0
+${If} $0 != "admin" ;Require admin rights on NT4+
+        messageBox mb_iconstop "Administrator rights required!"
+        setErrorLevel 740 ;ERROR_ELEVATION_REQUIRED
+        quit
+${EndIf}
+!macroend
+
+Function remoteWriteInfo
+    nsDialogs::Create 1018
+    Pop $Dialog
+
+    ${If} $Dialog == error
+        Abort
+    ${EndIf}
+    ${NSD_CreateLabel} 0 0 100% 12u "Enter URL for remote write"
+    Pop $RemoteWriteLabel
+
+    ${NSD_CreateText} 0 13u 100% 12u ""
+    Pop $RemoteWriteText
+
+    ${NSD_CreateLabel} 0 26u 100% 12u "Enter User for remote write"
+    Pop $RemoteWriteUserLabel
+
+    ${NSD_CreateText} 0 39u 100% 12u ""
+    Pop $RemoteWriteUserText
+
+    ${NSD_CreateLabel} 0 52u 100% 12u "Enter Password for remote write"
+    Pop $RemoteWritePasswordLabel
+
+    ${NSD_CreateText} 0 65u 100% 12u ""
+    Pop $RemoteWritePasswordText
+
+    nsDialogs::Show
+FunctionEnd
+
+
+Function remoteWriteInfoLeave
+    ${NSD_GetText}  $RemoteWriteText  $RemoteUrl
+    ${NSD_GetText}  $RemoteWriteUserText  $RemoteUser
+    ${NSD_GetText}  $RemoteWritePasswordText  $RemotePassword
+FunctionEnd
+ 
+Function enableWindowsExporter
+    nsDialogs::Create 1018
+    Pop $Dialog
+
+    ${If} $Dialog == error
+        Abort
+    ${EndIf}
+
+    ${NSD_CreateLabel} 0 0 100% 12u "Add and Enable Windows Exporter"
+    Pop $EnableExporterLabel
+
+    ${NSD_CreateCheckBox} 0 13u 100% 12u ""
+    Pop $EnableExporterCheck
+
+    nsDialogs::Show
+FunctionEnd
+
+Function enableWindowsExporterLeave
+    ${NSD_GetState} $EnableExporterCheck $EnableExporterValue
+FunctionEnd
+ 
+ 
+function .onInit
+    setShellVarContext all
+    !insertmacro VerifyUserIsAdmin
+functionEnd
```

Comment:
small nit: it seems like we're mixing the casing of keywords throughout the file, it'd be nice to be consistent (imo use `Function/FunctionEnd` since that seems to be how the NSIS docs reference them)